### PR TITLE
RGB color inversion and complementary color calculation added

### PR DESCRIPTION
**Pull request by brummer-simon**
I've added to the color library the inversion of a given rgb color as the calculation of the complementary color of a given rgb color

### CODE REVIEW

File: sys/color/color.c [hunk 1]
```diff
@@ -175,3 +176,29 @@ void color_rgb2str(const color_rgb_t *rgb, char* str)
...
+    max = (rgb->r > rgb->g) ? rgb->r : rgb->g;
+    max = (rgb->b > max) ? rgb->b : max;
+    min = (rgb->r < rgb->g) ? rgb->r : rgb->g;
+    min = (rgb->b < min) ? rgb->b : min;
+    val = max + min;
+
+    comp_rgb->r = val - rgb->r;
+    comp_rgb->g = val - rgb->g;
+    comp_rgb->b = val - rgb->b;
+}
```
Comment (authmillenon) [inline sys/color/color.c]: Some whitespace error here.

**Comment by authmillenon** [comment]
May I ask why?

**Comment by brummer-simon** [comment]
A little side project of mine is an API for rgb ledstrip controlling. Some additional color calculatons could be useful to generate nice effects.

**Comment by authmillenon** [comment]
I can accept that :-)

File: sys/color/color.c [hunk 2]
```diff
@@ -175,3 +176,27 @@ void color_rgb2str(const color_rgb_t *rgb, char* str)
     tmp = rgb->b & 0x0F;
     str[5] = (tmp > 9) ? ('A' - 10 + tmp) : ('0' + tmp);
 }
+
+void color_rgb_invert(const color_rgb_t *rgb, color_rgb_t *inv_rgb)
+{
+    inv_rgb->r = 255 - rgb->r;
+    inv_rgb->g = 255 - rgb->g;
+    inv_rgb->b = 255 - rgb->b;
+}
```
Comment (authmillenon) [inline sys/color/color.c]: This might be more ROM efficient, when inlined... (should be easy to test this, too ;-P)

File: sys/include/color.h [hunk 3]
```diff
@@ -104,6 +105,22 @@ void color_str2rgb(const char *str, color_rgb_t *color);
...
+ */
+void color_rgb_invert(const color_rgb_t *rgb, color_rgb_t *inv_rgb);
+
+/**
+ * @brief Calculate the complementary color of a given rgb color
+ *
+ * @param[in] rgb       Input rgb color.
+ * @param[out] comp_rgb Output rgb color, result of the complementary color calculation.
+ */
+void color_rgb_complementary(const color_rgb_t *rgb, color_rgb_t *comp_rgb)
```
Comment (authmillenon) [inline sys/include/color.h]: There is a difference between the inversion and the complementary color in additive color mixing ^^?

Comment (authmillenon) [inline sys/include/color.h]: If this is the HSL complementary (the code looks a little bit like it), I would really prefer to also add a HSL type and conversion functions (and use them in this function).

Comment (authmillenon) [inline sys/include/color.h]: Though… this might not be as efficient... Mh...

Comment (brummer-simon) [inline sys/include/color.h]: For main colors in rgb space inversion and complementary is actually the same, with secoundary colors it is not. The code is not the HSL complementary, I basicly do what adobe does xD

https://helpx.adobe.com/de/illustrator/using/adjusting-colors.html#WS714a382cdf7d304e7e07d0100196cbc5f-6288a

Comment (authmillenon) [inline sys/include/color.h]: Could you add that link, or at least refer to it as Adobe Complementary or something like that in the doc of this function then, please?

**Comment by brummer-simon** [comment]
Added Unittests, according to the calculation example from the adobe site and referenced the site.

File: sys/include/color.h [hunk 4]
```diff
@@ -104,6 +105,30 @@ void color_str2rgb(const char *str, color_rgb_t *color);
...
+/**
+ * @brief Calculate the complementary color of a given rgb color.
+ *
+ * @note                Complementary color calculation according to adobe illustator calculations.
+ *                      See https://helpx.adobe.com/illustrator/using/adjusting-colors.html
+ *
+ * @param[in] rgb       Input rgb color.
+ * @param[out] comp_rgb Output rgb color, result of the complementary color calculation.
+ */
+void color_rgb_complementary(const color_rgb_t *rgb, color_rgb_t *comp_rgb);
```
Comment (authmillenon) [inline sys/include/color.h]: Now that you have added unittests: Documenting the precondition that both `rgb` and the out variable can't be `NULL` would be great for both of these functions.

Comment (brummer-simon) [inline sys/include/color.h]: Done

File: sys/include/color.h [hunk 5]
```diff
@@ -104,6 +105,30 @@ void color_str2rgb(const char *str, color_rgb_t *color);
  */
 void color_rgb2str(const color_rgb_t *rgb, char *str);
 
+/**
+ * @brief Invert a given rgb color
+ *
+ * @param[in] rgb       Input rgb color, that should be converted. Must be NOT NULL
```
Comment (authmillenon) [inline sys/include/color.h]: Could you please use the `@pre` tag for that (either additionally or alternatively; your choice ;-))

File: sys/include/color.h [hunk 6]
```diff
@@ -104,6 +105,30 @@ void color_str2rgb(const char *str, color_rgb_t *color);
...
+}
+
+/**
+ * @brief Calculate the complementary color of a given rgb color.
+ *
+ * @note                Complementary color calculation according to adobe illustator calculations.
+ *                      See https://helpx.adobe.com/illustrator/using/adjusting-colors.html
+ *
+ * @param[in] rgb       Input rgb color. Must be NOT NULL
+ * @param[out] comp_rgb Output rgb color, result of the complementary color calculation. Must be NOT NULL
```
Comment (authmillenon) [inline sys/include/color.h]: dito

**Comment by authmillenon** [comment]
One last thing about documentation, then I'm happy :-)

**Comment by brummer-simon** [comment]
@pre added

**Comment by authmillenon** [comment]
ACK and go.

**Pull request merged**